### PR TITLE
emscripten: fix caveats

### PR DESCRIPTION
**Pull request by dnicolson**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

This pull request replaces the `caveats` instructions with a `post_install` block. When `emcc` is run for the first time the config file is created at `/usr/local/opt/emscripten/libexec/.emscripten`, this is now done automatically and the environment variables are added.

Previously the test would fail once the config file had been created as it expected a file argument.

### CODE REVIEW

**Comment by SMillerDev** [comment]
If it's in libexec by default, can't we set the values and remove the caveat?

**Comment by dnicolson** [comment]
Perhaps, `emcc` needs to be run first though. Should that be part of the install process?

**Comment by SMillerDev** [comment]
It could be added to a postinstall block if the file doesn't exist yet.

**Comment by dnicolson** [comment]
I can't find an example of a postinstall block, is there any documentation?

**Comment by gromgit** [comment]
See, for instance, `post_install` in https://github.com/Homebrew/homebrew-core/blob/master/Formula/postgresql.rb. The PostgreSQL formula creates some working directories and an initial database cluster if it doesn't already exist.

File: Formula/emscripten.rb [hunk 1]
```diff
@@ -92,18 +92,28 @@ def install
...
-      Manually set LLVM_ROOT to
-        #{opt_libexec}/llvm/bin
-      and BINARYEN_ROOT to
-        #{Formula["binaryen"].opt_prefix}
-      in ~/.emscripten after running `emcc` for the first time.
-    EOS
+  def post_install
+    system bin/"emcc"
+    text = File.read("#{libexec}/.emscripten")
+    new_contents = text.gsub(/^(LLVM_ROOT.*)/, "#\\1\nLLVM_ROOT = \"#{opt_libexec}/llvm/bin\"\\2")
```
Comment (SMillerDev) [inline Formula/emscripten.rb:98]: Could you use `inreplace` here?

File: Formula/emscripten.rb [hunk 2]
```diff
@@ -92,18 +92,28 @@ def install
...
+      #include <stdio.h>
+      int main()
+      {
+        printf("Hello World!");
+        return 0;
+      }
+    EOS
+
+    system bin/"emcc", "test.c", "-o", "test.js", "-s", "NO_EXIT_RUNTIME=0"
+    assert_equal "Hello World!", shell_output("node test.js").chomp
```
Comment (SMillerDev) [inline Formula/emscripten.rb:117]: 👍 🥳

**Review by SMillerDev — APPROVED** [review]
Thanks @dnicolson  ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

**Pull request closed without merging**